randobot/handler.py

```diff
@@ -65,18 +67,27 @@ class RandoHandler(RaceHandler):
     stop_at = ['cancelled', 'finished']
     max_status_checks = 50
 
-    def __init__(self, zsr, **kwargs):
+    def __init__(self, zsr, midos_house, **kwargs):
         super().__init__(**kwargs)
         self.zsr = zsr
+        self.midos_house = midos_house
 
     def should_stop(self):
         goal_name = self.data.get('goal', {}).get('name')
         goal_is_custom = self.data.get('goal', {}).get('custom', False)
-        return (
-            (goal_name == 'Random settings league' and not goal_is_custom) # handled by https://github.com/fenhl/rslbot
-            or (goal_name == '3rd Multiworld Tournament' and goal_is_custom) # handled by https://github.com/midoshouse/midos.house
-            or super().should_stop()
-        )
+        if goal_is_custom:
+            if goal_name == 'Random settings league':
+                return True # handled by https://github.com/fenhl/rslbot
+        else:
+            with contextlib.suppress(Exception): # if anything goes wrong, assume Mido's House is down and we should handle the room
+                query = gql.gql("""
+                    query {
+                        goalNames
+                    }
+                """)
+                if goal_name in self.midos_house.execute(query)['goalNames']:
```

Comment:
Ensure that this HTTP request has a suitable timeout value (not sure what the default timeout would be here)

Reply:
According to <https://gql.readthedocs.io/en/stable/modules/gql.html#gql.Client>, the default is 10 seconds.